tests/test_resolve.py

```diff
@@ -491,6 +494,44 @@ def test_nonexistent():
     assert raises(ResolvePackageNotFound, lambda: r.install(['numpy 1.5']))
 
 
+def test_timestamps_and_deps():
+    # If timestamp maximization is performed too early in the solve optimization,
+    # it will force unnecessary changes to dependencies. Timestamp maximization needs
+    # to be done at low priority so that conda is free to consider packages with the
+    # same version and build that are most compatible with the installed environment.
+    index2 = {Dist(key): value for key, value in iteritems(index)}
+    index2[Dist('mypackage-1.0-hash27_0.tar.bz2')] = IndexRecord(**{
+        'build': 'hash27_0',
+        'build_number': 0,
+        'depends': ['python 2.7*'],
+        'name': 'mypackage',
+        'requires': ['python 2.7*'],
+        'version': '1.0',
+        'timestamp': 1,
+    })
+    index2[Dist('mypackage-1.0-hash33_0.tar.bz2')] = IndexRecord(**{
+        'build': 'hash33_0',
+        'build_number': 0,
+        'depends': ['python 3.3*'],
+        'name': 'mypackage',
+        'requires': ['python 3.3*'],
+        'version': '1.0',
+        'timestamp': 0,
+    })
+    r = Resolve(index2)
+    installed1 = r.install(['python 2.7*', 'mypackage'])
+    print([k.dist_name for k in installed1])
+    assert any(k.name == 'python' and k.version.startswith('2.7') for k in installed1)
+    assert any(k.name == 'mypackage' and k.build == 'hash27_0' for k in installed1)
+    installed2 = r.install(['python 3.3*', 'mypackage'])
+    assert any(k.name == 'python' and k.version.startswith('3.3') for k in installed2)
+    assert any(k.name == 'mypackage' and k.build == 'hash33_0' for k in installed2)
+    installed3 = r.install(['mypackage'], r.install(['python 2.7*']))
+    assert installed1 == installed3
+    installed4 = r.install(['mypackage'], r.install(['python 3.3*']))
+    assert installed2 == installed4
+
```

Comment:
I will try to write a test that captures the weird hdf5 case, too.  For my memory's sake, the issue was:

- user has pytables built against hdf5 1.10
- user issues update command for pytables
- newest pytables built for hdf5 1.8 and 1.10, but 1.8 build has slightly newer timestamp
- upgrade ends up choosing pytables built against hdf5 1.8, forcing downgrade of hdf5 1.8